### ISSUE TITLE
[a11y]: IBM DataSpreadsheet Accessibility issues

### ISSUE THREAD

**Issue by AustinGitHub** [carbon-design-system/ibm-products#6668]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.51.0-rc.0

### React version

18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Using this tool 
https://github.com/IBMa/equal-access?tab=readme-ov-file#usage

Expectation is for data spreadsheet to be compliant, our due date is by June 2025. Please let me know if they are already compliant and with what version. 

### Reproduction/example

NA

### Steps to reproduce

Run the accessibility checker tool in the devtools against the data spreadsheet UI. This is the properties we have set on our component in case this is necessary to know. 

```jsx
<DataSpreadsheet
                columns={headerData ?? []}
                data={rowData ?? []}
                cellSize="lg"
                onDataUpdate={updateDataInfo}
                onSelectionAreaChange={selectedData}
                onActiveCellChange={selectedActiveCell}
                defaultEmptyRowCount={1}
                onColDrag={onColumnDragDrop}
                readOnlyTable={readOnlyTable}
                renderRowHeaderDirection="left"
                renderRowHeader={(currentData && !currentData.isReadOnly && !isUserViewerRole) && buildComponent}
                id="spreadsheet--id"
                key="spreadsheet--id"
                selectAllAriaLabel="DataSpreadsheetSelectAll" // Required
                spreadsheetAriaLabel="DataSpreadsheet" // Required
            />}
```

Please close this out with screenshot of the accessibility checker showing 0 violations, thanks!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by AustinGitHub** [comment]
![Image](https://github.com/user-attachments/assets/2928b62b-d392-40e3-be1b-db5855bb60cf)

**Comment by elycheea** [comment]
@AustinGitHub Since this component is currently in experimental and we are looking to replace this with a third-party/vendor solution, fixes for the `DataSpreadsheet` are currently not priorities for our team. If you’re able to contribute the fix, that would probably be the best option at the moment.
